src/Testcontainers.MsSql/MsSqlBuilder.cs

```diff
@@ -131,12 +131,20 @@ private MsSqlBuilder WithUsername(string username)
     /// </remarks>
     private sealed class WaitUntil : IWaitUntil
     {
-        private readonly string[] _command = { "/opt/mssql-tools/bin/sqlcmd", "-Q", "SELECT 1;" };
-
         /// <inheritdoc />
         public async Task<bool> UntilAsync(IContainer container)
         {
-            var execResult = await container.ExecAsync(_command)
+            var hasMsSql18Tools = await container.ExecAsync(new[] { "[", "-f" , "/opt/mssql-tools18/bin/sqlcmd", "]" })
+                .ConfigureAwait(false);
+     
+            string[] command = [ 
+                hasMsSql18Tools.ExitCode == 0 ? "/opt/mssql-tools18/bin/sqlcmd" : "/opt/mssql-tools/bin/sqlcmd",
```

Comment:
Consider moving the paths to named variables.

Reply:
Added them as constants and exposed a new internam method on the container to avoid duplication.